nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/avro/BlobPropertyMapper.java

```diff
@@ -0,0 +1,117 @@
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.model.impl.primitives.BlobProperty;
+import org.nuxeo.runtime.RuntimeServiceException;
+import org.nuxeo.runtime.avro.AvroMapper;
+import org.nuxeo.runtime.avro.AvroService;
+
+/**
+ * @since 10.2
+ */
+public class BlobPropertyMapper extends AvroMapper<BlobProperty, Object> {
+
+    public BlobPropertyMapper(AvroService service) {
+        super(service);
+    }
+
+    @Override
+    public Object fromAvro(Schema schema, Object input) {
+        switch (schema.getType()) {
+        case NULL:
+            if (input == null) {
+                return null;
+            }
+            throw new NonNullValueException();
+        case UNION:
+            for (Schema sub : schema.getTypes()) {
+                try {
+                    return service.fromAvro(sub, BlobProperty.class, input);
+                } catch (NonNullValueException e) {
+                    // ignore
+                }
+            }
+            throw new RuntimeServiceException(CANNOT_MAP_FROM + schema.getType());
+        case RECORD:
+            try {
+                GenericRecord record = (GenericRecord) input;
+                String mimeType = (String) record.get(service.encodeName("mime-type"));
+                String encoding = (String) record.get("encoding");
+                byte[] bytes = ((ByteBuffer) record.get("data")).array();
+                Blob b = Blobs.createBlob(bytes, mimeType, encoding);
+                b.setFilename((String) record.get("name"));
+                b.setDigest((String) record.get("digest"));
+                return b;
+            } catch (IOException e) {
+                throw new RuntimeServiceException(CANNOT_MAP_FROM + schema.getType(), e);
+            }
+        default:
+            throw new RuntimeServiceException(CANNOT_MAP_FROM + schema.getType());
+        }
+    }
+
+    @Override
+    public Object toAvro(Schema schema, BlobProperty input) {
+        switch (schema.getType()) {
+        case NULL:
+            if (input == null) {
+                return null;
+            }
+            throw new RuntimeServiceException(CANNOT_MAP_TO + schema.getType());
+        case UNION:
+            for (Schema s : schema.getTypes()) {
+                try {
+                    return service.toAvro(s, input);
+                } catch (RuntimeServiceException e) {
+                    // ignore
+                }
+            }
+            throw new RuntimeServiceException(CANNOT_MAP_TO + schema.getType());
+        case RECORD:
+            GenericRecord record = new GenericData.Record(schema);
+            for (Field f : schema.getFields()) {
+                if ("data".equals(f.name())) {
+                    Blob blob = (Blob) input.getValue();
+                    try {
+                        record.put(f.name(), ByteBuffer.wrap(blob.getByteArray()));
+                    } catch (IOException e) {
+                        e.printStackTrace();
```

Comment:
Does it make sense to rethrow the exception as `UncheckedIOException` or `NuxeoException` ?
If not, can you log the exception ?

Reply:
doen